Objects/longobject.c

```diff
@@ -4491,38 +4494,59 @@ long_rshift1(PyLongObject *a, Py_ssize_t wordshift, digit remshift)
 {
     PyLongObject *z = NULL;
     Py_ssize_t newsize, hishift, i, j;
-    digit lomask, himask;
+    digit lomask, himask, omitmark;
 
-    if (Py_SIZE(a) < 0) {
-        /* Right shifting negative numbers is harder */
-        PyLongObject *a1, *a2;
-        a1 = (PyLongObject *) long_invert(a);
-        if (a1 == NULL)
-            return NULL;
-        a2 = (PyLongObject *) long_rshift1(a1, wordshift, remshift);
-        Py_DECREF(a1);
-        if (a2 == NULL)
-            return NULL;
-        z = (PyLongObject *) long_invert(a2);
-        Py_DECREF(a2);
+    /* for a positive integrate x
+         -x >> m = -(x >> m)      when x is two's pow
+                 = -(x >> m) -1   otherwise */
+
+    if (IS_MEDIUM_VALUE(a)) {
+        stwodigits sval;
+        if (wordshift > 0)
+            return get_small_int((sdigit)Py_SIZE(a) >> 1);
+
+        sval = medium_value(a) >> remshift;
+        if (IS_SMALL_INT(sval))
```

Comment:
[PEP 7](https://www.python.org/dev/peps/pep-0007/) says all newly-written code should include `{braces}` around blocks.